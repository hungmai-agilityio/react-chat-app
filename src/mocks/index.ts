export const List_item_chat = [
  {
    id: 'sade3123-45435ft-553ksd-6754534',
    avatar: 'assets/images/mock-image.jpg',
    name: 'John',
    isActive: true,
    isGroup: true,
    message: 'Hello',
    time_stamp: new Date('2024-07-01T15:12:05.791Z').toString(),
    path: '/john'
  },
  {
    id: 'sa14345ft-553ksd-67095450j-4849',
    avatar: '',
    name: 'Marketing',
    isGroup: true,
    isGroupChat: true,
    message: 'Hello team',
    time_stamp: '2024-05-24T15:58:05.791Z',
    path: '/marketing'
  },
  {
    id: '53ksd-67w4435567435ft-55ft-553k4534',
    avatar: '',
    name: 'Ban',
    isActive: false,
    isGroup: true,
    message: 'Hello world',
    time_stamp: '2020-10-01T15:37:01.791Z',
    path: '/ban'
  }
];

export const List_item_user = [
  {
    id: '53ksd-67w4435545677-11445ds4534',
    avatar: 'assets/images/mock-image.jpg',
    name: 'John',
    isActive: true,
    path: '/john'
  },
  {
    id: '53ksd-67w4435567-43567-342341-13k4534',
    avatar: '',
    name: 'Max',
    isActive: true,
    path: '/max'
  },
  {
    id: '53ksd-67whu7du435ft-5ti5ft-553ken4534',
    avatar: '',
    name: 'Ban',
    isActive: false,
    path: '/max'
  }
];

export const memberInfo = [
  {
    id: '701796f0-0253-11ef-804b-d90ac4375227',
    name: 'Alice',
    avatar: '',
    isActive: true
  },
  {
    id: '7017234f0-0253-11ef-804b-d90ac4375227',
    name: 'Bob',
    avatar: 'assets/images/mock-image.jpg',
    isActive: true
  },
  {
    id: '701796f0-0253-11ef-804b-dr544375227',
    name: 'Charlie',
    avatar: '',
    isActive: false
  }
];

export const memberInfoSolo = [
  {
    id: '701796f0-0253-11ef-804b-d90ac4375227',
    name: 'Alice',
    avatar: '',
    isActive: true
  },
  {
    id: '7017234f0-0253-11ef-804b-d90ac4375227',
    name: 'Bob',
    avatar: 'assets/images/mock-image.jpg',
    isActive: true
  }
];

export const mockUsers = [
  {
    id: '8a741820-0253-11ef-804b-d90ac4375227',
    userName: 'Mary',
    email: 'mary@abc.com',
    password: 'Test12345',
    created_at: new Date(),
    updated_at: new Date()
  },
  {
    id: 'e7d1d160-078f-11ef-9ebd-d50e40f719a7',
    userName: 'Faker',
    email: 'faker@abc.com',
    password: 'Faker123',
    created_at: new Date(),
    updated_at: new Date()
  }
];

export const mockProfiles = {
  '701796f0-0253-11ef-804b-d90ac4375227': {
    id: '701796f0-0253-11ef-804b-d90ac4375227',
    user_id: '701796f0-0253-11ef-804b-d90ac4375227',
    avatar: '',
    phone: '123-456-7890'
  },
  '8a741820-0253-11ef-804b-d90ac4375227': {
    id: '8a741820-0253-11ef-804b-d90ac4375227',
    user_id: '8a741820-0253-11ef-804b-d90ac4375227',
    avatar: '',
    phone: '987-654-3210'
  },
  'e7d1d160-078f-11ef-9ebd-d50e40f719a7': {
    id: 'e7d1d160-078f-11ef-9ebd-d50e40f719a7',
    user_id: 'e7d1d160-078f-11ef-9ebd-d50e40f719a7',
    avatar: '',
    phone: '456-789-0123'
  }
};

export const mockDropdownItem = [
  { label: 'Item 1', value: 'item1' },
  { label: 'Item 2', value: 'item2' },
  { label: 'Item 3', value: 'item3' }
];

export const mockMessage = {
  id: '43274s-102-11ef-n741820-95h5hts325677j4562',
  message: 'Hello',
  sender: '8a741820-0253-11ef-804b-d90ac4375227',
  time_stamp: '2024-04-24T15:58:05.791Z',
  roomId: '97651-07i2-11ef-n741820-95h5hts32543543e2',
  isEdit: false
};

export const mockMessageEdit = {
  id: '93274s-29011ef-9ebd-95h5hts3256701h311',
  message: 'Hi',
  sender: '7017234f0-0253-11ef-804b-d90ac4375227',
  time_stamp: '2024-04-25T15:58:05.791Z',
  roomId: '97651-07i2-11ef-n741820-95h5hts32543543e2',
  isEdit: true
}
