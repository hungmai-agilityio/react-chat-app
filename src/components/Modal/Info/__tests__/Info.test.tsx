import { memberInfo } from '@/mocks';
import { fireEvent, render, screen } from '@testing-library/react';
import ModalInfo from '..';

const defaultProps = {
  member: memberInfo,
  avatar: 'assets/images/mock-image.jpg',
  name: 'Bob',
  count: 3,
  currentUserId: '701796f0-0253-11ef-804b-d90ac4375227'
};
const onClickMock = jest.fn();

describe('ModalInfo', () => {
  it('renders correctly with default props', () => {
    render(<ModalInfo {...defaultProps} />);

    expect(screen.getByText('Members (3)')).toBeInTheDocument();
    memberInfo.forEach((user) => {
      expect(screen.getByText(user.name)).toBeInTheDocument();
    });
  });

  it('renders correctly with click event', () => {
    const { getByText } = render(
      <ModalInfo {...defaultProps} isGroup onClick={onClickMock} />
    );

    const buttonElement = getByText('Leave chat');

    fireEvent.click(buttonElement);

    expect(onClickMock).toHaveBeenCalled();
  });
});

it('should render match to snapshot', () => {
  const { container } = render(<ModalInfo {...defaultProps} />);
  expect(container).toMatchSnapshot();
});
