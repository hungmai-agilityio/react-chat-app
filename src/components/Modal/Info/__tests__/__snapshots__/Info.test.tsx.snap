// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render match to snapshot 1`] = `
<div>
  <div
    class="mt-3 relative"
  >
    <div
      class="flex justify-center"
    >
      <img
        alt="Bob"
        class="object-cover font-bold w-avatar-lg h-avatar-lg text-8xl"
        data-testid="avatar"
        src="assets/images/mock-image.jpg"
      />
    </div>
    <div
      class="my-5 mx-4 h-modal-md"
    >
      <p>
        Members 
        (3)
      </p>
      <div
        class="h-modal-md overflow-y-auto scrollbar"
      >
        <div>
          <div
            class="flex items-center justify-between hover:bg-blue-50 w-full"
          >
            <div
              class="text-sm px-3 w-20 flex items-center gap-3 w-full h-item"
            >
              <div
                class="relative"
              >
                <div
                  class="text-primary bg-gray-300 flex items-center justify-center font-semibold rounded-full my-0 mx-auto w-avatar-sm h-avatar-sm text-lg"
                  data-testid="avatar"
                >
                  A
                </div>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-circle absolute text-gray-400 border-xs border-white rounded-full top-6 left-7 text-green-500"
                  data-icon="circle"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              <div
                class="flex flex-col truncate"
              >
                <p>
                  Alice
                   
                  <span
                    class="text-green-600"
                  >
                    (You)
                  </span>
                </p>
              </div>
            </div>
          </div>
          <div
            class="flex items-center"
          >
            <div
              class="border-b border-gray-300 w-full"
            />
            <div
              class="border-b border-gray-300 w-full"
            />
          </div>
        </div>
        <div>
          <div
            class="flex items-center justify-between hover:bg-blue-50 w-full"
          >
            <div
              class="text-sm px-3 w-20 flex items-center gap-3 w-full h-item"
            >
              <div
                class="relative"
              >
                <img
                  alt="Bob"
                  class="object-cover font-bold rounded-full my-0 mx-auto w-avatar-sm h-avatar-sm text-lg"
                  data-testid="avatar"
                  src="assets/images/mock-image.jpg"
                />
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-circle absolute text-gray-400 border-xs border-white rounded-full top-6 left-7 text-green-500"
                  data-icon="circle"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              <div
                class="flex flex-col truncate"
              >
                <p>
                  Bob
                   
                </p>
              </div>
            </div>
          </div>
          <div
            class="flex items-center"
          >
            <div
              class="border-b border-gray-300 w-full"
            />
            <div
              class="border-b border-gray-300 w-full"
            />
          </div>
        </div>
        <div>
          <div
            class="flex items-center justify-between hover:bg-blue-50 w-full"
          >
            <div
              class="text-sm px-3 w-20 flex items-center gap-3 w-full h-item"
            >
              <div
                class="relative"
              >
                <div
                  class="text-primary bg-gray-300 flex items-center justify-center font-semibold rounded-full my-0 mx-auto w-avatar-sm h-avatar-sm text-lg"
                  data-testid="avatar"
                >
                  C
                </div>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-circle absolute text-gray-400 border-xs border-white rounded-full top-6 left-7"
                  data-icon="circle"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              <div
                class="flex flex-col truncate"
              >
                <p>
                  Charlie
                   
                </p>
              </div>
            </div>
          </div>
          <div
            class="flex items-center"
          >
            <div
              class="border-b border-gray-300 w-full"
            />
            <div
              class="border-b border-gray-300 w-full"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
